### DISCUSSION TITLE
how to run a laravel app locally ?

### DISCUSSION THREAD

**Discussion by lakoud** [Schweinepriester/achievement-testarea#55] [Q&A]
how to run a laravel app locally ?

**Comment by rubikproxy** [comment]
To run a Laravel app locally, follow these steps:

1. **Install PHP and Composer**:  
   Ensure you have PHP (version 8.0 or higher) and Composer installed on your system.

2. **Download Laravel Project**:  
   Clone the Laravel project repository or download the files into your local directory.

3. **Install Dependencies**:  
   Navigate to the project folder in your terminal and run:  
   ```bash
   composer install
   ```

4. **Set Up Environment File**:  
   Create a `.env` file by copying the `.env.example` file:  
   ```bash
   cp .env.example .env
   ```

5. **Generate Application Key**:  
   Run the following command to generate the app key:  
   ```bash
   php artisan key:generate
   ```

6. **Configure the Database**:  
   Update the `.env` file with your local database credentials.

7. **Run Database Migrations** (if applicable):  
   Execute the migrations to create database tables:  
   ```bash
   php artisan migrate
   ```

8. **Start the Development Server**:  
   Launch the Laravel application with the built-in server:  
   ```bash
   php artisan serve
   ```

9. **Access the Application**:  
   Open your browser and visit `http://127.0.0.1:8000` to see your Laravel app running locally.
